webrecorder/browsertrix
# Crawl config detail views

WIP https://github.com/webrecorder/browsertrix-cloud/issues/89

- Adjust details in list view for improved readability
- New detail view page with configuration, schedule and crawl details

TODOs recorded here https://github.com/webrecorder/browsertrix-cloud/issues/100

### Manual testing
1. Run app and go to crawl template list page. Click on a crawl template name, verify it takes you to the detail page
2. Choose or create a template with more than 3 seed URLs. Verify the detail view has a "show more" button that displays all the URLs
3. Click "Advanced Configuration". It should show the entire JSON `config` object
3. Scroll to "Crawls" section and click "Run now". Verify crawl starts

### Screenshots
**Enhancements to last/next crawl detail in list view*
<img width="673" alt="Screen Shot 2022-01-24 at 6 52 43 PM" src="https://user-images.githubusercontent.com/4672952/150903908-273002d4-c074-4958-b0a1-3e5fbd58775f.png">

**New detail view**
<img width="856" alt="Screen Shot 2022-01-24 at 6 53 08 PM" src="https://user-images.githubusercontent.com/4672952/150903966-b0f01a8e-3e4d-44a5-b874-bdc6d9bf4a55.png">

**Expanded seeds**
<img width="582" alt="Screen Shot 2022-01-24 at 6 53 44 PM" src="https://user-images.githubusercontent.com/4672952/150903997-bfcdf89b-782a-4b1b-887e-a8c5d4a4e9b6.png">

**Expanded "Advanced Configuration"**
<img width="579" alt="Screen Shot 2022-01-24 at 6 54 04 PM" src="https://user-images.githubusercontent.com/4672952/150904023-666f7e25-1b63-47ed-b593-8810fa2d54e7.png">

**Formatted schedule**
<img width="846" alt="Screen Shot 2022-01-24 at 6 54 49 PM" src="https://user-images.githubusercontent.com/4672952/150904065-0d0cbd46-ebf6-4e4a-a911-10b8627c1695.png">

**Banner in detail view when crawl is running**
<img width="870" alt="Screen Shot 2022-01-24 at 6 54 21 PM" src="https://user-images.githubusercontent.com/4672952/150904043-061abac7-03ba-470b-9a21-79fcf609c6d6.png">

**Crawls section when no crawl is currently running**
<img width="840" alt="Screen Shot 2022-01-24 at 7 07 54 PM" src="https://user-images.githubusercontent.com/4672952/150904116-228630a1-ef8a-4b38-8b91-af42006568a5.png">
